### PR TITLE
Bump LLVM

### PR DESCRIPTION
**Pull request by adam-smnk**
Fixes after bump:
  - Update IR syntax
  - Resolve broken implicit conversions when initializing pass options
  - Remove deprecated partial bufferization
  - Fix APInt ctor assertion - use zero for undefined length vector read

### CODE REVIEW

**Comment by rengolin** [comment]
Unless there's an urgent need to fix these issues, let's wait until #980 and #987 are merged.

**Comment by adam-smnk** [comment]
No rush, we're just pretty outdated.

**Review by rengolin — APPROVED** [review]

**Comment by adam-smnk** [comment]
Rebased + solved conflicts, bumped again + updated error message

**Pull request merged**
